src/nft-card.ts

```diff
@@ -50,123 +71,101 @@ export class NftCard extends LitElement {
-  @property({type: String})  public width = this.horizontal ? '670px' : '388px'
-  @property({type: String})  public height = this.horizontal ? '250px' : '560px'
-  @property({type: String})  public network = 'mainnet'
-
-  @property({type: String})  private flippedCard = false
-  @property({type: Boolean}) private loading = true
-
-  @property({type: Object})  public asset
-  @property({type: Object})  public traitData = {}
-  @property({type: String})  public account
-
-  private getNetwork() {
-    switch (this.network) {
-      case 'mainnet': return Network.Main
-      case 'rinkeby': return Network.Rinkeby
     }
-  }
-
-  constructor() {
-    super()
-  }
-
-  /**
-   * ConnectedCallback - Invoked when a component is added to the document’s DOM.
-   * Grabs data from the OpenSea SDK and populates data objects to be passed to
-   * child components.
-   */
-  public async connectedCallback() {
-    super.connectedCallback()
-    this.provider =
-      typeof web3 !== 'undefined'
-        ? window.web3.currentProvider
-        : Web3.providers.HttpProvider('https://mainnet.infura.io')
-
-    this.web3 = new Web3(this.provider)
-    this.seaport = new OpenSeaPort(this.provider, {networkName: this.getNetwork()})
-    await this.connectWallet()
-    this.asset = await this.seaport.api.getAsset(this.contractAddress, this.tokenId);
-
-    [this.account] = await this.web3.eth.getAccounts()
-
-    // TODO: Check if account exists
-    if(this.account) {
-        this.asset.isOwnedByAccount = this.asset.owner.address.toLowerCase() === this.account.toLowerCase()
+
+    /**
+     * ConnectedCallback - Invoked when a component is added to the document’s DOM.
+     * Grabs data from the OpenSea SDK and populates data objects to be passed to
+     * child components.
+     */
+    public async connectedCallback() {
+        super.connectedCallback()
+
+        this.seaport = new OpenSeaPort(this.provider, {networkName: this.getNetwork()})
+        await this.connectWallet()
+
+        ;[this.account] = await this.web3.eth.getAccounts()
+
+        await this.getAsset(this.contractAddress, this.tokenId)
+
+        if (this.account) {
+            this.asset.isOwnedByAccount = this.asset.owner.address.toLowerCase() === this.account.toLowerCase()
+        }
+
+        this.traitData = {
+            traits: this.asset.traits,
+            collectionTraits: this.asset.collection.traitStats
+        }
+
+        // We got the data so we are done loading
+        this.loading = false
+
+        // Tell the component to update with new state
+        await this.requestUpdate()
+
+        // Watch for the account to change then update state of component
+        window.ethereum.on('accountsChanged', function (accounts) {
+            [this.account] = accounts
+            console.log('accountsChanged', accounts)
+        })
+    }
+
+    public async buyEvent() {
+
+        const order = await this.seaport.api.getOrder({
```

Comment:
by default, we order sell orders by createdDate descending. so this should give the lowest price order, but not necessary the same one fetched from getAsset, in this.asset.sellOrders i think. we should just use that one directly

Reply:
@alexanderatallah  are you saying we should use the order object from `this.asset.sellOrders` instead of fetching the order here?

Reply:
Something like this?
```js
const order = this.asset.sellOrders[0]
await this.seaport.fulfillOrder({order, accountAddress: this.account})
```